### PR TITLE
Live: enable per-detector analysis flags

### PR DESCRIPTION
**Pull request by titodalcanton**
I do some refactoring of the data reader code in `pycbc_live`, and add the ability to use different analysis flags for different detectors. This is necessary for Virgo in O3.

Note that this patch will require a change in the current PyCBC Live configuration.

Needs testing.

### CODE REVIEW

**Comment by titodalcanton** [comment]
@ahnitz I still need to test this more carefully, but the approach is finalized, so ready for your review.

**Comment by titodalcanton** [comment]
This is now tested with fake frame files carrying various combinations of states and DQ vetoes. @ahnitz ready for your review.

File: bin/pycbc_live [hunk 1]
```diff
@@ -287,6 +285,54 @@ class LiveEventManager(object):
                 if store_psd[ifo] is not None:
                     store_psd[ifo].save(fname, group='%s/psd' % ifo)
 
+def init_data_reader(ifo, args, maxlen):
```
Comment (ahnitz) [inline bin/pycbc_live]: Should this be a class method of StrainBuffer?

**Comment by ahnitz** [comment]
@titodalcanton One minor question, otherwise, this seems fine.

File: bin/pycbc_live (hunk 1 above)
Comment (titodalcanton) [reply bin/pycbc_live]: Makes sense, moved to a `from_cli` class method.

**Pull request merged**